src/Navigation.re

```diff
@@ -27,7 +27,7 @@ external addListener:
         "key": string,
       },
       "context": string,
-      "type": [@bs.string] [ | `didBlur | `didFocus | `willBlur | `willFocus],
+      "type": [ | `didBlur | `didFocus | `willBlur | `willFocus],
```

Comment:
Interesting, bs warn about this, but this is send to js, so we must have this annotation. @cknitt any idea why bs yells sometimes when it shouldn't?

Reply:
`[@bs.string]` will only work directly on arguments of external functions, not on fields of objects passed to external functions or things like that.

Therefore the usage here is incorrect - the type of the `type` field within the callback parameter is `string`, it does not get automatically converted to a polymorphic variant. So it is good that we got a warning.

Actually bs.string should be used for the second parameter of the addListener function though.